packages/discussions/src/utils/channel-permission.ts

```diff
@@ -0,0 +1,262 @@
+import { IGroup } from "@esri/arcgis-rest-types";
+import {
+  AclCategory,
+  AclSubCategory,
+  IChannelAclPermission,
+  IChannelAclPermissionDefinition,
+  IDiscussionsUser,
+  Role,
+} from "../types";
+import { CANNOT_DISCUSS } from "./constants";
+import { isOrgAdmin } from "./platform";
+
+type PermissionsByAclCategoryMap = {
+  [key in AclCategory]?: IChannelAclPermission[];
+};
+
+export class ChannelPermission {
+  private readonly ALLOWED_GROUP_MEMBER_TYPES = ["owner", "admin", "member"];
+  private readonly ALLOWED_ROLES_FOR_POSTING = [
+    Role.WRITE,
+    Role.READWRITE,
+    Role.MANAGE,
+    Role.MODERATE,
+    Role.OWNER,
+  ];
+
+  private isChannelAclEmpty: boolean;
+  private permissionsByCategory: PermissionsByAclCategoryMap;
+
+  constructor(channelAcl: IChannelAclPermission[]) {
+    this.isChannelAclEmpty = channelAcl.length === 0;
+    this.permissionsByCategory = {};
+
+    channelAcl.forEach((permission) => {
+      const { category } = permission;
+      this.permissionsByCategory[category]?.push(permission) ||
+        (this.permissionsByCategory[category] = [permission]);
+    });
+  }
+
+  canPostToChannel(user: IDiscussionsUser) {
+    if (this.aclAllowsAnyUserToPost()) {
+      return true;
+    }
+
+    if (!this.isUserLoggedIn(user)) {
```

Comment:
This is a nit, but this would be better if we didn't have to grok the `!`;  so instead, something like `isUserUnauthenticated`?

Reply:
Yep I like it 👍